hypothesis-python/docs/details.rst

```diff
@@ -446,6 +449,29 @@ and should be rewritten as:
             return result
 
 
+An alternative hook is provided for use by test runner extensions such as
+:pypi:`pytest-trio`, which cannot use the ``execute_example`` method.
+This is **not** recommended for end-users - it is better to write a complete
+test function directly, perhaps by using a decorator to perform the same
+transformation before applying :func:`@given <hypothesis.given>`.
```

Comment:
Given the discussion we had in the context of `pytest-trio`, I think we need to better document what the decorator needs to handle. In particular we need to note that the replaced function must handle both args and kwargs to pass through to the underlying test function.

Reply:
Done.